File: frontend/src/types/transaction.ts
export interface Transaction {
  id: number;
  amount: number;
  description: string;
  type: string; // Backend returns as string
  date: string;
  createdAt: string;
}

export interface CreateTransactionDto {
  amount: number;
  description: string;
  type: TransactionType;
  date: string; // ISO date string format
}

export enum TransactionType {
  Income = 0,
  Expense = 1,
}

export interface BalanceResponse {
  balance: number;
}

export interface ApiErrorResponse {
  message: string;
  error?: string;
}

File: SimpleAccounting.API/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace SimpleAccounting.API.Models
{
    public class Transaction
    {
        public int Id { get; set; }
        
        [Required]
        public decimal Amount { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Description { get; set; } = string.Empty;
        
        [Required]
        public TransactionType Type { get; set; }
        
        [Required]
        public DateTime Date { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }

    public enum TransactionType
    {
        Income,
        Expense
    }
}